transaction/core/src/tx_summary.rs

```diff
@@ -86,50 +86,26 @@
 //! based on the TxPrefix. So the host computer can gain no advantage by lying
 //! to the device in this way.
 
-use crate::{
-    tx::{TxOut, TxPrefix},
-    CompressedCommitment, MaskedAmount,
-};
 use alloc::{collections::BTreeMap, vec::Vec};
-use mc_crypto_digestible::Digestible;
-use mc_crypto_keys::CompressedRistrettoPublic;
-use mc_util_zip_exact::{zip_exact, ZipExactError};
-use prost::Message;
-use serde::{Deserialize, Serialize};
-use zeroize::Zeroize;
 
-/// A subset of the data in a Tx which enables efficient verification (e.g. by a
-/// HW wallet) of the inputs and outputs of a transaction being signed.
-#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
-pub struct TxSummary {
-    /// The outputs which will be added to the blockchain as a result of this
-    /// transaction
-    #[prost(message, repeated, tag = "1")]
-    pub outputs: Vec<TxOutSummary>,
+use crate::{tx::TxPrefix, TxOut};
 
-    /// Data in the summary associated to each real input
-    #[prost(message, repeated, tag = "2")]
-    pub inputs: Vec<TxInSummary>,
-
-    /// Fee paid to the foundation for this transaction
-    #[prost(uint64, tag = "3")]
-    pub fee: u64,
+use mc_crypto_keys::CompressedRistrettoPublic;
+use mc_crypto_ring_signature::CompressedCommitment;
+use mc_util_zip_exact::{zip_exact, ZipExactError};
 
-    /// Token id for the fee output of this transaction
-    #[prost(uint64, tag = "4")]
-    pub fee_token_id: u64,
+pub use mc_transaction_types::tx_summary::*;
 
-    /// The block index at which this transaction is no longer valid.
-    #[prost(uint64, tag = "5")]
-    pub tombstone_block: u64,
+/// [Summarize] trait for constructing [TxSummary] object from foreign types
+pub trait Summarize {
+    /// Make a TxSummary for a given TxPrefix and pseudo-output commitments
```

Comment:
It looks to me that we are taking the `TxSummary::new` function and reorganizing it as a trait, implemented on a pair of references.

I can't see any reasons why this is better and I see a lot of reasons why it's worse. Can we back out if this change?

Reply:
I see, to break dependencies?

Reply:
Okay, can we maybe do this like

`impl TryFrom<(&...,&...)> for TxSummary`?

I think that's more idiomatic and its more helpful to someone reading rustdoc documentation

Reply:
so i need to move the `TxSummary` objects to types to share em between transaction/{core/extra/summary} but the actual logic should stay in core (to be moved to summary), so the trait is a trick to allow impl on a foreign type...

open to alternatives? or i could refactor the `Summarize` trait to act on the `TxSummary` object?

```rust
pub trait Summarize {
  fn new(&TxPrefix, &[CompressedCommitment]) -> Result<TxSummary, ZipExactError> { .. }
}
```

Reply:
Right -- what I said doesnt work with orphan rules in the way you're shooting for.

What if we spelled the trait `TxSummaryNew` similar to Tony arcieri's `AeadNew` trait? I feel like calling it `Summarize` isn't very clear. Lots of things can be "summarized" but this is really specifically meant to be the constructor for TxSummary.

Reply:
yeah good call, swapped to `TxSummaryNew` ^_^